account_asset_management/models/account_move.py

```diff
@@ -159,17 +163,26 @@ class AccountMoveLine(models.Model):
     _inherit = "account.move.line"
 
     asset_profile_id = fields.Many2one(
-        comodel_name="account.asset.profile", string="Asset Profile",
+        comodel_name="account.asset.profile",
+        string="Asset Profile",
+        compute="_compute_asset_profile",
+        store=True,
+        readonly=False,
+        copy=True,
     )
     asset_id = fields.Many2one(
         comodel_name="account.asset", string="Asset", ondelete="restrict",
     )
 
-    @api.onchange("account_id")
-    def _onchange_account_id(self):
-        if self.account_id.asset_profile_id:
-            self.asset_profile_id = self.account_id.asset_profile_id
-        super()._onchange_account_id()
+    @api.depends("account_id", "asset_id")
+    def _compute_asset_profile(self):
+        for rec in self:
+            if rec.account_id.asset_profile_id and not rec.asset_id:
+                rec.asset_profile_id = rec.account_id.asset_profile_id
+            elif rec.asset_id:
+                rec.asset_profile_id = rec.asset_id.profile_id
+            else:
+                rec.asset_profile_id = False
```

Comment:
This is incorrect and causing a bug. Fixed in 424ac791dddc27a0ea161c45dc8977356f353c3c